src/util/color_spaces.ts

```diff
@@ -1,20 +1,34 @@
-import Color from './color';
-
-import {number as interpolateNumber} from './interpolate';
-
-type LABColor = {
-    l: number;
-    a: number;
-    b: number;
-    alpha: number;
-};
-
-type HCLColor = {
-    h: number;
-    c: number;
-    l: number;
-    alpha: number;
-};
+/**
+ * @param r Red component 0..1
+ * @param g Green component 0..1
+ * @param b Blue component 0..1
+ * @param alpha Alpha component 0..1
+ */
+export type RGBColor = [r: number, g: number, b: number, alpha: number];
```

Comment:
I tend to say the previous implementation was better in terms of debugging, finding references etc as the new definition is basically all a 4 numbers array which is extremely error prone.

Reply:
These data types do not go beyond color interpolation - in other words, they are only used when called by the `interpolate.color` function which instead initially takes colors as instances of the `Color` class and which also returns colors as instances of the `Color` class. So I don't think that declaring this data as objects gives much in the way of improving the debuggability of the code. 
On the other hand, as arrays, they are much more versatile and allow, for example, for reuse of code to interpolate arrays 
https://github.com/maplibre/maplibre-style/blob/e679ba7e2bf31c5d7ed99110bae654c5d257beb2/src/util/interpolate.ts#L37
https://github.com/maplibre/maplibre-style/blob/e679ba7e2bf31c5d7ed99110bae654c5d257beb2/src/util/interpolate.ts#L70
instead of the need to create such code specifically for each object of different shape:
https://github.com/maplibre/maplibre-style/blob/2408da0c626110c8525b98d6a05b9985f6be885c/src/util/interpolate.ts#L17-L24
As for error proneness: because of the aforementioned isolation of this code from the rest (only used internally by `interpolate.color`) and tests written for them I don't believe that using arrays here is a bad idea.

Reply:
I would consider adding a `toArray` method to allow interpolation and not have the entire API based on 4 numbers array. The reason I'm saying that, is that if you try to debug this code, add a console log etc, there's no indication if the array holds RGB HSL or LAB, it all looks the same, and it's confusing and can cause very hard to fix bugs, I've seen this happen over and over in my carrier.
If the main reason is for the interpolation code to be easier to write then have a utility method at that entry point.
I see that in most cases you spread the array into its components, so this won't change much I think, but it will make the API easier to understand.

Reply:
I understand your concerns, however, please respect my own judgment and my experience.